rest_lights.cpp

```diff
@@ -1667,28 +1715,34 @@ int DeRestPluginPrivate::setWarningDeviceState(const ApiRequest &req, ApiRespons
     {
         if (alert == "none")
         {
-            task.options = 0x00; // Warning mode 0 (no warning), No strobe
+            task.options = 0x00; // Warning mode 0 (no warning), No strobe, Low sound
             task.duration = 0;
         }
         else if (alert == "select")
         {
-            task.options = isSmokeDetector
-              ? 0x12  // Warning mode 2 (fire), Strobe
-              : 0x14; // Warning mode 1 (burglar), Strobe
+            task.options = 0x17; // Warning mode 1 (burglar), Strobe, Very high sound
             task.duration = 1;
         }
         else if (alert == "lselect")
         {
-            task.options = isSmokeDetector
-              ? 0x12  // Warning mode 2 (fire), Strobe
-              : 0x14; // Warning mode 1 (burglar), Strobe
+            task.options = 0x17; // Warning mode 1 (burglar), Strobe, Very high sound
             task.duration = onTime > 0 ? onTime : 300;
         }
         else if (alert == "blink")
         {
-            task.options = 0x04; // Warning mode 0 (no warning), Strobe
+            task.options = 0x04; // Warning mode 0 (no warning), Strobe, Low sound
             task.duration = onTime > 0 ? onTime : 300;
         }
+        else if (alert == "all")
+        {
+            // FXIME: Dirty hack to send a network-wide broadcast to activate all sirens.
+            task.req.dstAddress().setNwk(deCONZ::BroadcastAll);
+            task.req.setDstAddressMode(deCONZ::ApsNwkAddress);
+            task.req.setTxOptions(0);
+            task.req.setDstEndpoint(0xFF);
+            task.options = 0x17; // Warning mode 1 (burglar), Strobe, Very high sound
+            task.duration = onTime > 0 ? onTime : 1;
+        }

```

Comment:
We have two sirens installed in the office now, this can be fun, totally need to test this out.

Reply:
Make sure to wear protective ear plugs.

What type of sirens do you have?  Do they support groups?

Reply:
Absolutely :) We have one from Heiman and one from Tuya (branded as wesmartify).

![image](https://user-images.githubusercontent.com/383386/81055869-db7a0880-8ec9-11ea-9c2b-e1d83debe757.png)

Reply:
Nice, mains-powered, battery-backed-up and OTAU as well.  Do they publish firmware updates?

You might want to whitelist this siren, so the API plugin creates a ZHAAlarm `/sensors` resource.  I think that's needed for automatic enrolment.  Also, it will expose the level of the backup battery.

After refactoring the `/lights` API, I want to do `/groups`, supporting groups of warning devices, and window covering devices.  Simplest way is bluntly to expose `action.open` on every group; smarter way would be to keep track what type of devices are in the group (what would also be needed e.g. not to expose `action.xy` on a group of dimmable lights).  I might need to introduce another attribute for `alert`, to distinguish between alerting lights and alerting sirens.

Alternatively, use homogenous groups of `type` `WarningGroup` and `WindowCoveringGroup` (next to `LightGroup`).  That would block heterogeneous scenes, though.  I've yet to see a siren that supports scenes, but both the IKEA FYRTUR and `lumi.curtain` support it.